model/src/main/scala/cromwell/pipeline/model/wrapper/VersionValue.scala

```diff
@@ -0,0 +1,50 @@
+package cromwell.pipeline.model.wrapper
+
+import cats.data.{ NonEmptyChain, Validated }
+import cats.implicits.catsStdShowForString
+import cromwell.pipeline.model.validator.Wrapped
+import play.api.libs.json.Format
+
+import scala.util.matching.Regex
+
+final class VersionValue private (override val unwrap: Int) extends AnyVal with Wrapped[Int]
+
+object VersionValue extends Wrapped.Companion {
+  type Type = Int
+  type Wrapper = VersionValue
+  type Error = String
+  type VersionGroup = (VersionValue, VersionValue, VersionValue)
+  type StringGroup = (String, String, String)
+  val pattern = "^[0-9]+$"
+  val fullVersionPattern: Regex = "^v([0-9]+)\\.([0-9]+)\\.([0-9]+)$".r
+  implicit val versionNumberFormat: Format[VersionValue] = wrapperFormat
+  def increment(value: VersionValue): VersionValue = create(value.unwrap + 1)
+  def getFullVersion(value: String): ValidationResult[VersionGroup] =
+    validateFullVersion(value) match {
+      case Validated.Valid((v1, v2, v3)) => Validated.Valid((create(v1.toInt), create(v2.toInt), create(v3.toInt)))
+      case Validated.Invalid(errors)     => Validated.Invalid(errors)
+    }
+  protected def validateFullVersion(value: String): ValidationResult[StringGroup] = Validated.cond(
+    value.matches(fullVersionPattern.regex),
+    value match {
+      case fullVersionPattern(v1, v2, v3) => (v1, v2, v3)
+    },
+    NonEmptyChain.one(s"Format of version name: 'v(int).(int).(int)', but got: $value")
+  )
```

Comment:
What are you doing?
I told that `VersionValue` is a **single** version. You must not parse a full version here, this the responsibility of the `PipelineVersion` method!

Reply:
As I get it from a previous message, this should be in micro type. In my previous commit validate form of full version was in PipelineVersion

Reply:
Microtype `VersionValue` must contain information about parsing a single value into a `VersionValue` and nothing more.
`PipelineVersion` must know that it consists of three `VersionValue`s.
In your previous commit, `PipelineVersion` also knew that `VersionValue` consists of digits, which is wrong.
Now `VersionValue` knows that `PipelineVersion` consists of three `VersionValues`, which is also wrong.

Reply:
`PipelineVersion` must take a full string, validate that it matches the following regex (or something similar, idk)
`^v([^.]*).([^.]*).([^.]*)$`
If it matches then it passes each of these three elements into `VersionValue.fromString` method. That's all.